### PR TITLE
fix: (TNLT-6168) - Android - display carousel in modal 

### PR DESCRIPTION
**Pull request by kamiranoff**
- https://nidigitalsolutions.jira.com/browse/TNLT-2130
- Display carousel in Android if multiple images in the article

Before:
Screenshot_1610389152

After:
Screenshot_1610389255
Screenshot_1610472473

### CODE REVIEW

File: packages/article-lead-asset/src/article-lead-asset.js [hunk 1]
```diff
@@ -71,7 +70,7 @@ const ArticleLeadAsset = ({
     <View style={style}>
       <LeadAsset
         aspectRatio={getRatio(crop.ratio)}
-        caption={<Caption {...caption} />}
+        caption={caption}
```
Comment (kamiranoff) [inline packages/article-lead-asset/src/article-lead-asset.js:73]: This is not as flexible as before but `Caption` was the only component used throughout the project.

I could also add the possibility back to put a component, which will result in fewer test changes.

File: packages/image/src/modal-image/modal-image.js [hunk 2]
```diff
@@ -32,21 +33,38 @@ class ModalImage extends Component {
...
-    return React.cloneElement(caption, { style });
+    if (this.state.currentIndex === 0) {
+      return (
+        <Caption style={style} credits={caption.credits} text={caption.text} />
+      );
+    }
+
+    if (
+      images[this.state.currentIndex - 1] &&
+      images[this.state.currentIndex - 1].attributes
```
Comment (c-j-j) [inline packages/image/src/modal-image/modal-image.js:56]: You can use optional chaining here, i.e. `images[this.state.currentIndex - 1]?.attributes`

File: packages/image/src/modal-image/modal-image.js [hunk 3]
```diff
@@ -32,21 +33,38 @@ class ModalImage extends Component {
...
-    const { caption } = this.props;
+    const { caption, images } = this.props;
     const style = isTablet ? tabletCaptionStyles : captionStyles;
 
     if (!caption) {
       return null;
     }
 
-    return React.cloneElement(caption, { style });
+    if (this.state.currentIndex === 0) {
```
Comment (c-j-j) [inline packages/image/src/modal-image/modal-image.js:48]: Could destructure `currentIndex` out of `this.state` as you're using it in a few places.

File: packages/utils/src/get-extra-images-content.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+type Article = {
```
Comment (c-j-j) [inline packages/utils/src/get-extra-images-content.ts:1]: We have started to create some domain types here - so I think you can replace the `content` here with `ArticleContent` https://github.com/newsuk/times-components-native/blob/f4bdb911e7ffeca4ad3c95c89b319bec211f032a/packages/types/domain-types.ts#L75

File: packages/utils/src/get-extra-images-content.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      relativeHeight: number;
+      relativeHorizontalOffset: number;
+      relativeVerticalOffset: number;
+      relativeWidth: number;
+      url: string;
+    };
+  }[];
+};
+
+export const getExtraImagesContent = (article: Article) => {
```
Comment (c-j-j) [inline packages/utils/src/get-extra-images-content.ts:18]: Minor naming suggestion: `getInlineImages` may be a bit simpler?

**Review by c-j-j — APPROVED** [review]
Looks good to me - only a few minor comments for you to peruse

File: packages/utils/src/get-extra-images-content.ts (hunk 5 above)
Comment (kamiranoff) [reply packages/utils/src/get-extra-images-content.ts:18]: Ah yes. I chose `getExtraImagesContent` to show that it was returning more than just images.

File: packages/article-lead-asset/src/article-lead-asset.js (hunk 1 above)
Comment (aronshamash) [reply packages/article-lead-asset/src/article-lead-asset.js:74]: I think render props are a good thing but if only ever `Caption` this makes sense

File: packages/image/image.showcase.js [hunk 6]
```diff
@@ -140,7 +140,7 @@ export default {
       component: () => (
         <View style={{ width: 300 }}>
           <Image
-            aspectRatio={1 / 1}
```
Comment (aronshamash) [inline packages/image/image.showcase.js:143]: 😆

**Review by aronshamash — APPROVED** [review]
looks really good

**Review by c-j-j — APPROVED** [review]

**Pull request merged**